build.yaml

```diff
@@ -196,13 +232,26 @@ common_cli:
     # SSL on our caching box.
     # The --trusted-host option is only available with pip 7.
     #
-    parse_logs pip install --upgrade pip
+    if [ "\$(is_ubuntu)" == "true" ] || [ "\$(is_centos)"  == "true" ]; then
+        PIP_ADDITIONAL_OPTIONS=" -i \${PIP_INDEX_URL} --trusted-host \${TRUSTED_HOST}"
+        PARSE_LOGS="parse_logs "
+    fi
+    if [ "\$(is_osx)" == "true" ]; then
+      # we can't use our pip caching layer on our OSX cloudbees boxes
+      # as we're unable run the slave_setup plugin that injects the environment
+      # variables to the slave.
+      # So we just go upstream for the python modules.
+      PIP_ADDITIONAL_OPTIONS=""
+      PARSE_LOGS=""
+    fi
+
```

Comment:
because the escape codes didn't provide colour on the output.
it can be tackled on a different ticket.
my preference would be not to use awk but something easier on the eyes